docs/DeveloperGuide.adoc

```diff
@@ -126,31 +126,37 @@ NOTE: The lifeline for `DeleteCommandParser` should end at the destroy marker (X
-.Class Diagram of the Suggestion Engine Component (still WIP)
+.Class Diagram of the Suggestion Engine Component
 image::SuggestionClassDiagram.png[]
 
 *API* :
-link:{repoURL}blob/master/src/main/java/com/notably/logic/suggestion/SuggestionEngine.java[`SuggestionEngine.java`]
+link:{repoURL}/blob/master/src/main/java/com/notably/logic/suggestion/SuggestionEngine.java[`SuggestionEngine.java`]
 
 `SuggestionEngine` gives users the meaning of the command they input and a list of notes suggestions that they want to
-open/ delete/ search/ edit.
-
-1. `Logic` uses the `SuggestionEngineImpl` class, which implements `SuggestionEngine` interface, to parse the user command.
-2. According to the command the user inputs, `SuggestionEngineImpl` will create a `XYZSuggestionCommandParser` object
-which implements `SuggestionCommandParser` interface.
-3. This results in the creation of `XYZSuggestionCommand` object which implements `SuggestionCommand` interface.
-`XYZSuggestionCommand` is then executed by the `LogicManager`.
-4. The command execution can affect the `Model` (e.g. `open` command will set the response text in the model as
-"Open a note" and store a list of `SuggestionItem` in the model).
-5. The UI will then be able to retrieve the response text and list of `SuggestionItem` from the model to be displayed
+open, delete, or search.
+
+1. `Logic` uses the `SuggestionEngine` class, to handle the user input.
+2. According to the command the user inputs, `SuggestionEngine` will create a `XYZSuggestionArgHandler` or `ABCSuggestionHandler` object
+which implements `SuggestionArgHandler` and `SuggestionHandler` interface respectively. `XYZSuggestionArgHandler` are for commands that
+require argument parsing, i.e. `open`, `delete`, `search`, `new`, whereas `ABCSuggestionHandler` are for commands that do not require argument
+parsing, i.e. `edit`, `exit`, `help`.
+3. If `SuggestionArgHandler` object is created: the `responseText` in the `Model` will be updated. This case will also result in the
+creation of `XYZSuggestionGenerator` object (except for `new` command) which implements `SuggestionGenerator` interface.
+`XYZSuggestionGenerator` is then executed by the `SuggestionEngine`.
+4. If `SuggestionHandler` object is created: the `responseText` in the `Model` will be updated.
+5. The `Model` could be affected in 2 ways:
+* Update `responseText` of the `Model` (by the `SuggestionHandler`): for instance, the input `open /` will set the `responseText` in the `Model` as
+"Open a note".
+* Store a list of `SuggestionItem` in the `Model` (by the `SuggestionGenerator`).
+6. The UI will then be able to retrieve the `responseText` and list of `SuggestionItem` from the `Model` to be displayed
 to the user.
 
 Given below is the Sequence Diagram for interactions within the `Logic` and `Suggestion` component for the `execute("open /")` API call.
 
 .Interactions Inside the Logic and Suggestion Component for the `open /` Command
 image::SuggestionSequenceDiagram.png[]
```

Comment:
Although this is an old diagram, I feel that you can update this to remove the correction section and just focus on the suggestions because it's quite large (starting to become unreadable) and unnecessary for the purpose of explaining the suggestions engine.